source/config/__init__.py

```diff
@@ -19,37 +19,6 @@
 import easeOfAccess
 import winKernel
 
-def validateConfig(configObj,validator,validationResult=None,keyList=None):
-	"""
-	@deprecated: Add-ons which need this should provide their own implementation.
-	"""
-	import warnings
-	warnings.warn("config.validateConfig deprecated. Callers should provide their own implementation.",
-		DeprecationWarning, 2)
-	if validationResult is None:
-		validationResult=configObj.validate(validator,preserve_errors=True)
-	if validationResult is True:
-		return None #No errors
-	if validationResult is False:
-		return "Badly formed configuration file"
-	errorStrings=[]
-	for k,v in validationResult.iteritems():
-		if v is True:
-			continue
-		newKeyList=list(keyList) if keyList is not None else []
-		newKeyList.append(k)
-		if isinstance(v,dict):
-			errorStrings.extend(validateConfig(configObj[k],validator,v,newKeyList))
-		else:
-			#If a key is invalid configObj does not record its default, thus we need to get and set the default manually 
-			defaultValue=validator.get_default_value(configObj.configspec[k])
-			configObj[k]=defaultValue
-			if k not in configObj.defaults:
-				configObj.defaults.append(k)
-			errorStrings.append("%s: %s, defaulting to %s"%(k,v,defaultValue))
-	return errorStrings
-
-#: @deprecated: Use C{conf.validator} instead.
 val = Validator()

```

Comment:
um, fail on my part.


On 6/9/2016 6:01 PM, James Teh wrote:
>
> In source/config/__init__.py 
> <https://github.com/nvaccess/nvda/pull/5932#discussion_r66542894>:
>
> > -		if v is True:
> > -			continue
> > -		newKeyList=list(keyList) if keyList is not None else []
> > -		newKeyList.append(k)
> > -		if isinstance(v,dict):
> > -			errorStrings.extend(validateConfig(configObj[k],validator,v,newKeyList))
> > -		else:
> > -			#If a key is invalid configObj does not record its default, thus we need to get and set the default manually
> > -			defaultValue=validator.get_default_value(configObj.configspec[k])
> > -			configObj[k]=defaultValue
> > -			if k not in configObj.defaults:
> > -				configObj.defaults.append(k)
> > -			errorStrings.append("%s: %s, defaulting to %s"%(k,v,defaultValue))
> > -	return errorStrings
> > -
> > -#: @deprecated: Use C{conf.validator} instead.
> >  val = Validator()
>
> You've removed the deprecation note here, but not the variable. We may 
> as well remove this variable too and instead make ConfigManager 
> instantiate Validator directly.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/pull/5932/files/ad494d45126e9b22df7a0056f84aba9dfafdcef7#r66542894>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AFGivWQGa2KLS-F7ddaC3KKtp4Hhdp0bks5qKKlrgaJpZM4IY4RQ>.
>

-- 
------------------------------------------------------------------------


    Derek Riemer

  * Department of computer science, third year undergraduate student.
  * Proud user of the NVDA screen reader.
  * Open source enthusiast.
  * Member of Bridge Cu
  * Avid skiier.

Websites:
Honors portfolio <http://derekriemer.com>
Awesome little hand built weather app! 
<http://django.derekriemer.com/weather/>

email me at derek.riemer@colorado.edu <mailto:derek.riemer@colorado.edu>
Phone: (303) 906-2194